.github/workflows/codeql.yml

```diff
@@ -0,0 +1,75 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
```

Comment:
question(blocking): this doesn't look like it will run on every branch, it looks like it will run on any branch targeting the main branch in a PR. Is my assumption correct?

suggestion: it'd be nice to have it ran on every PR even if we're not targeting the main branch.

Reply:
Referenced from [here](https://github.com/github/codeql-action/issues/462)

Reply:
praise: thanks, looks good, resolving this conversation